### ISSUE TITLE
JAWs stops reading legend elements after ~840 characters

### ISSUE THREAD

**Issue by noahhi** [FreedomScientific/VFO-standards-support#482]
### Summary

> If you are using a fieldset/legend construct and have a long passage of text in the legend element (for example, a terms and conditions text), JAWs will only read the legend up until a seemingly random cutoff around 840 characters (or ~1020 characters including spaces) at which point it will jump ahead and start reading the remaining content of the fieldset. I was not able to locate any JAWs setting related to this behavior so it seems to be a bug.
> 
> Example:
> 1. Go to https://jsbin.com/moniduyaxe/edit?html,output
> 2. Tab to the 'yes' button

### Expected result

> I expect JAWs to read the legend text (a passage from Alice in Wonderland) in its entirety.

### Actual result

> JAWs stops reading at around 840 characters (~1020 characters including spaces). 

### Additional Information
The exact character count where it cuts off tends to vary slightly depending on the text and the browser 

#### JAWS version and build number
JAWS Version 2020.2008.24 ILM

 
#### Operating System and version
Windows 10 Enterprise

#### Browser and version:
Tested in Chrome, Firefox, and Internet Explorer

**Comment by JAWS-test** [comment]
I would not consider this a JAWS error. A group label should be as concise as possible, since it is always output by the screen reader at the beginning (see also [WCAG 2.4.6](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html)). If the group label is too long, the name, role and value of the focused control cannot be perceived efficiently. In my opinion, a group label should have a maximum of 80 characters. Longer text content should be inserted as text and linked as description if necessary (aria-describedby)

**Comment by frex65** [comment]
To qualify @JAWS-test's comment above, if the element after the lengthy legend were, for example, a checkbox, users would be forced to listen to the whole legend before being able to discern whether or not the checkbox was checked, which would be inefficient and frustrating. Group labels should be as concise as possible while still providing a reasonable description of the group's purpose.

**Comment by BenKeyFSI** [comment]
The free preview time for the original JS Bin sample has expired. In addition the original JS Bin sample has a bug, the border of the field set runs right through the middle of the legend.

An improved sample is now available at https://cdpn.io/TPG/debug/VwKNVaG.

**Comment by BenKeyFSI** [comment]
While I agree with the comments made by @JAWS-test and @frex65 in principle, there is a JAWS bug that is causing the truncation.

**Comment by BenKeyFSI** [comment]
Reported as Bug 123389 {JAWS stops reading legend elements after ~840 characters} in the Vispero Bugzilla.
